File: Python/compiler/tokens.py
from enum import Enum, unique

#creando enumeracion de tokens
@unique
class Token(Enum):
	OpenBrace =  "{"
	CloseBrace = "}"
	OpenParen = "("
	CloseParen = ")"
	Semicolon = ";"
	IntKeyword = "int"
	CharKeyword = "char"
	ReturnKeyword = "return"
	BitwiseComplement = "~"
	LogicalNegation = "!"
	Plus = "+"
	Minus = "-"
	Multiplication = "*"
	Division = "/"

@unique
class UnaryOp(Enum):
	BitwiseComplement = Token.BitwiseComplement
	LogicalNegation = Token.LogicalNegation

@unique
class BinaryOp(Enum):
	Addition = Token.Plus
	Subtraction = Token.Minus
	Multiplication = Token.Multiplication
	Division = Token.Division

File: src/tokens.ml
type token =
	| OpenBrace
	| CloseBrace
	| OpenParen
	| CloseParen
	| Semicolon 
	| IntKeyword
	| MainKeyword
	| ReturnKeyword
	| Id of string
	| Int of int 
	| Error

let getInt tok =
	match tok with
	Int i -> i

let getId tok =
	match tok with
	Id str -> str